pkg/networkservice/common/roundrobin/server.go

```diff
@@ -44,37 +44,33 @@ func NewServer() networkservice.NetworkServiceServer {
 }
 
 func (s *selectEndpointServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
-	ctx, err := s.withClientURL(ctx, request.GetConnection())
-	if err != nil {
-		return nil, err
+	if clienturlctx.ClientURL(ctx) != nil {
+		return next.Server(ctx).Request(ctx, request)
 	}
-	return next.Server(ctx).Request(ctx, request)
-}
-
-func (s *selectEndpointServer) Close(ctx context.Context, conn *networkservice.Connection) (*empty.Empty, error) {
-	// TODO - we should remember the previous selection here.
-	ctx, err := s.withClientURL(ctx, conn)
-	if err != nil {
-		return nil, err
-	}
-	return next.Server(ctx).Close(ctx, conn)
-}
+	candidates := discover.Candidates(ctx)
 
-func (s *selectEndpointServer) withClientURL(ctx context.Context, conn *networkservice.Connection) (context.Context, error) {
-	if clienturlctx.ClientURL(ctx) == nil {
-		candidates := discover.Candidates(ctx)
+	for i := 0; i < len(candidates.Endpoints); i++ {
```

Comment:
Why no iterate till end of context deadline?

Reply:
ok, Denis answered. We had round robin per id, so we will iterate over all endpoints.